> LINK : fatal error LNK1104: cannot open file 'python313_d.lib' [D:\a\cpython\cpython\PCbuild\pythoncore.vcxproj]

Ok, I found the root issue and wrote a fix: PR gh-110725. The problem comes from PyAPI_FUNC(), PyAPI_DATA() and PyMODINIT_FUNC macros which have a different implementation depending if the 3 Py_BUILD_CORE macros are defined or not.

--- a/Doc/whatsnew/3.13.rst
+++ b/Doc/whatsnew/3.13.rst
@@ -928,6 +928,10 @@ Build Changes
 * Building CPython now requires a compiler with support for the C11 atomic
   library, GCC built-in atomic functions, or MSVC interlocked intrinsics.
 
+* The ``_stat`` C extension is now built with the :ref:`limited C API
+  <limited-c-api>`.
+  (Contributed by Victor Stinner in :gh:`85283`.)
+
 
 C API Changes
 =============

--- a/Misc/NEWS.d/next/Build/2023-08-29-15-05-09.gh-issue-85283.tlK7G7.rst
+++ b/Misc/NEWS.d/next/Build/2023-08-29-15-05-09.gh-issue-85283.tlK7G7.rst
@@ -1,0 +1,2 @@
+The ``_stat`` C extension is now built with the :ref:`limited C API
+<limited-c-api>`. Patch by Victor Stinner.

--- a/Modules/_stat.c
+++ b/Modules/_stat.c
@@ -11,6 +11,9 @@
  *
  */
 
+// Need limited C API version 3.13 for PyModule_Add() on Windows
+#define Py_LIMITED_API 0x030d0000
+
 #include "Python.h"
 
 #ifdef HAVE_SYS_TYPES_H

